sql/core/src/test/scala/org/apache/spark/sql/sources/PrunedScanSuite.scala

```diff
@@ -114,16 +114,15 @@ class PrunedScanSuite extends DataSourceTest with SharedSQLContext {
   testPruning("SELECT * FROM oneToTenPruned", "a", "b")
   testPruning("SELECT a, b FROM oneToTenPruned", "a", "b")
   testPruning("SELECT b, a FROM oneToTenPruned", "b", "a")
-  testPruning("SELECT b, b FROM oneToTenPruned", "b")
+  testPruning("SELECT b, b FROM oneToTenPruned", "b", "b")
+  testPruning("SELECT b as alias_b, b FROM oneToTenPruned", "b")
```

Comment:
When we do not return two columns, we will always have a `ProjectExec` above the table scan:
```
*Project [b#222 AS alias_b#240, b#222]
+- InMemoryTableScan [b#222]
```